### PR TITLE
Sandbox/Tenant Details - Config Properties - Tree Table

### PR DESCRIPTION
**Pull request by burzum619**
Included in this PR:
- Created new tree table component for use with configuration properties
- Created mappers for mapping tenants/sandboxes from data model to view model
- Integrating new tree table component into sandbox/tenant details 

TODOs in followup PR:
- Implement Save tenant/sandbox for configuration properties - need to ONLY include modified properties
- Fix filtering in treetable (filtering only modified properties from tree view)
- Integrate with "Create new sandbox" and "Create new tenant" screens - we need to pull default properties of current tenant to pre-populate config properties

### CODE REVIEW

File: webapp/src/main/webapp/src/app/admin/sandbox-tenants/model/configuration-property.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export class ConfigurationProperty {
   key: string;
   originalValue: string;
   value: string;
+  formControlName?: string;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/sandbox-tenants/model/configuration-property.ts]: is this used?

File: webapp/src/main/webapp/src/app/admin/sandbox-tenants/mapper/tenant.mapper.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  };
+}
+
+function mapConfigurationProperties(
+  configProperties: any,
+  overrides: any
+): any {
+  let groupedProperties = {};
+
+  forOwn(configProperties, (configGroup, groupKey) => {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/sandbox-tenants/mapper/tenant.mapper.ts]: haven't seen this one - is it like a forEach with hasOwnProperty?

Comment (burzum619) [reply webapp/src/main/webapp/src/app/admin/sandbox-tenants/mapper/tenant.mapper.ts]: thats exactly what it is

File: webapp/src/main/webapp/src/app/admin/sandbox-tenants/model/configuration-property.ts (hunk 1 above)
Comment (burzum619) [reply webapp/src/main/webapp/src/app/admin/sandbox-tenants/model/configuration-property.ts]: Yep, its used to bind the input fields in the tree table by a qualified name

**Pull request merged**